### PR TITLE
Add error log for flush task scheduling

### PR DESCRIPTION
**Pull request by sfc-gh-alhuang**
Add error log for flush task scheduling. This could happen when [getCachedMaxClientLagInMs](https://github.com/snowflakedb/snowflake-ingest-java/blob/eb87281a6a1f5ee3e2e959594bb571b40b770b90/src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java#L268C65-L268C92) throw cast exception when provided parameter is `int` not `long`.

### CODE REVIEW

File: src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java [hunk 1]
```diff
@@ -282,25 +282,37 @@ private void createWorkers() {
...
+          } catch (Exception e) {
+            String errorMessage =
+                String.format(
+                    "Failed to schedule a flush task, client=%s, exception=%s, detail=%s,"
+                        + " trace=%s.",
+                    this.owningClient.getName(),
+                    e.getClass().getName(),
+                    e.getMessage(),
+                    getStackTrace(e));
+            logger.logError(errorMessage);
```
Comment (sfc-gh-tzhang) [inline src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java:314]: Have you verify that we're able to see the error in the log?

Comment (sfc-gh-tzhang) [reply src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java:314]: Also could we re-throw it in the end?

File: src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java [hunk 2]
```diff
@@ -282,25 +282,37 @@ private void createWorkers() {
...
+          }
```
Comment (sfc-gh-tzhang) [inline src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java:315]: Consider also send it to SF similar to the logic above
```
                      if (this.owningClient.getTelemetryService() != null) {
                        this.owningClient
                            .getTelemetryService()
                            .reportClientFailure(this.getClass().getSimpleName(), errorMessage);
                      }
```

**Review by sfc-gh-tzhang — APPROVED** [review]
Left some comments, otherwise lgtm, thanks!

File: src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java (hunk 1 above)
Comment (sfc-gh-alhuang) [reply src/main/java/net/snowflake/ingest/streaming/internal/FlushService.java:314]: Verified and added re-throw logic.

**Pull request merged**